app/src/main/java/com/example/eric/mdbweather/MainActivity.java

```diff
@@ -0,0 +1,314 @@
+        searchView.setIconifiedByDefault(true);
+
+        searchView.setOnQueryTextListener(new SearchView.OnQueryTextListener() {
+
+            @Override
+            public boolean onQueryTextSubmit(String query) {
+
+                if(Geocoder.isPresent()){
+                    String location = query;
+                    Geocoder geoCoder = new Geocoder(getApplicationContext());
+                    List<Address> addresses= null;
+                    try {
+                        addresses = geoCoder.getFromLocationName(location, 5);
+                    } catch (IOException e) {
+                        e.printStackTrace();
+                    }
+                    latitude = addresses.get(0).getLatitude();
+                    longitude = addresses.get(0).getLongitude();
+                }
+
+                mSectionsPagerAdapter = new SectionsPagerAdapter(getSupportFragmentManager());
+                mViewPager = (ViewPager) findViewById(R.id.container);
+                mViewPager.setAdapter(mSectionsPagerAdapter);
+
+                return true;
+            }
+
+            @Override
+            public boolean onQueryTextChange(String newText) {
+                return false;
+            }
+        });
+
+        return true;
+    }
+
+    public static class PlaceholderFragment extends Fragment implements View.OnClickListener{
+        private static final String ARG_SECTION_NUMBER = "section_number";
+
+        private static TextView current_temp;
+        private static TextView current_time;
+        private static TextView cityNameView;
+        private static TextView current_summary;
+        private static TextView current_date;
+        private static TextView current_rain;
+        private static TextView poweredBy;
+
+        public PlaceholderFragment() {
+        }
+
+        public static PlaceholderFragment newInstance(int sectionNumber) {
+            PlaceholderFragment fragment = new PlaceholderFragment();
+            Bundle args = new Bundle();
+            args.putInt(ARG_SECTION_NUMBER, sectionNumber);
+            fragment.setArguments(args);
+            return fragment;
+        }
+
+        @Override
+        public void onClick(View v){
+            switch (v.getId()){
+                case R.id.poweredBy:
+                    Uri uri = Uri.parse(getString(R.string.dark_sky_powered));
+                    Intent intent = new Intent(Intent.ACTION_VIEW, uri);
+                    startActivity(intent);
+                    break;
+            }
+        }
+
+        @Override
+        public View onCreateView(LayoutInflater inflater, final ViewGroup container, Bundle savedInstanceState) {
+            int position = getArguments().getInt(ARG_SECTION_NUMBER);
+            final View rootView;
+
+            switch (position){
+                case 1:
+                    rootView = inflater.inflate(R.layout.fragment_main, container, false);
```

Comment:
This is a lot of stuff for one case, and makes the code really confusing to read. This should be divided into two or more different methods (maybe initialize everything in a private method that takes in the rootview, define the asynctask separately, and use the callable for http connection)